### PR TITLE
SO-3188: Add option for unpublished validation

### PR DESCRIPTION
**Pull request by norban**

### CODE REVIEW

File: core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequest.java [hunk 1]
```diff
@@ -216,6 +226,10 @@ public void setRuleIds(Collection<String> ruleIds) {
 		this.ruleIds = ruleIds;
 	}
 	
+	public void setFilterOptions(Map<String, Object> filterOptions) {
```
Comment (cmark) [inline core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequest.java]: Reduce visibility to `package private`.

File: core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequestBuilder.java [hunk 2]
```diff
@@ -28,20 +29,28 @@
...
 	private Collection<String> ruleIds;
 	
+	private Map<String, Object> filterOptions;
+	
 	public ValidateRequestBuilder setRuleIds(Collection<String> ruleIds) {
 		this.ruleIds = ruleIds;
 		return getSelf();
 	}
 	
+	public ValidateRequestBuilder setFilterOptions(Map<String, Object> filterOptions) {
```
Comment (cmark) [inline core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequestBuilder.java]: `ruleParameters` would be a better name for this request parameter, rename it pls.

File: core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/eval/GroovyScriptValidationRuleEvaluator.java [hunk 3]
```diff
@@ -38,13 +39,14 @@ public GroovyScriptValidationRuleEvaluator(Path validationResourcesDirectory) {
 	}
 	
 	@Override
-	public List<ComponentIdentifier> eval(BranchContext context, ValidationRule rule) throws Exception {
```
Comment (cmark) [inline core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/eval/GroovyScriptValidationRuleEvaluator.java:41]: Keep the `BranchContext` as a separate parameter to the scripts.

File: snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/ruleSnomedCommon1.groovy [hunk 4]
```diff
@@ -20,30 +26,33 @@ Iterable<Hits<String>> inactiveConceptBatches = searcher.scroll(Query.select(Str
...
+	
+	ExpressionBuilder invalidRelationshipExpression = Expressions.builder()
 			.filter(SnomedRelationshipIndexEntry.Expressions.active())
 			.should(SnomedRelationshipIndexEntry.Expressions.sourceIds(inactiveConceptIds))
 			.should(SnomedRelationshipIndexEntry.Expressions.typeIds(inactiveConceptIds))
 			.should(SnomedRelationshipIndexEntry.Expressions.destinationIds(inactiveConceptIds))
-			.build()
-
+			
+	if (params.containsKey(EffectiveTimes.UNSET_EFFECTIVE_TIME_LABEL)) {
```
Comment (cmark) [inline snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/ruleSnomedCommon1.groovy]: Use a different parameter name than `EffectiveTimes.UNSET_EFFECTIVE_TIME_LABEL` for Unpublished only validation

**Review by cmark — CHANGES_REQUESTED** [review]

**Comment by norban** [comment]
Requested changes have been applied

File: core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequestBuilder.java [hunk 5]
```diff
@@ -28,20 +29,28 @@
...
 	private Collection<String> ruleIds;
 	
+	private Map<String, Object> ruleParameters;
+	
 	public ValidateRequestBuilder setRuleIds(Collection<String> ruleIds) {
 		this.ruleIds = ruleIds;
 		return getSelf();
 	}
 	
+	public ValidateRequestBuilder setFilterOptions(Map<String, Object> ruleParameters) {
```
Comment (cmark) [inline core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/core/validation/ValidateRequestBuilder.java]: Please change the method name here and the parameter name in the ValidateRequest class as well.

**Review by cmark — CHANGES_REQUESTED** [review]

**Review by cmark — APPROVED** [review]
LGTM!

**Comment by zoliMolnar** [comment]
Thank you!

**Pull request merged**